### ISSUE TITLE
 Create a sprite sheet from multiple .aseprite files

### ISSUE THREAD

**Issue by thomasgoldstein** [aseprite/aseprite#3235]
Hello. This is for a feature request, which from what I've seen has not been requested and is not supported. Correct me if I'm wrong!

I have multiple .aseprite files that represent all of the possible animations for a given character. They’re like stand.aseprite, walk.aseprite, run.aseprite, etc. For the sake of simplicity, let’s call them anim1.aseprite, anim2.aseprite, anim3.aseprite, etc.

I’d like to create a single sprite sheet from all these files. The sprite sheet may have over hundreds of frames in total, so trying to maintain a single .aseprite containing all animations does not sound like a good idea.

In the generated sprite sheet, the first row would be all the frames from anim1.aseprite (maybe if they have a certain tag), the second row would be all the frames from anim2.aseprite, and so on, one row per .aseprite file. In my case, all frames will have the same size across animations, and each anim doesn’t necessarily have the same amount of frames, so it’s okay if each sprite sheet row doesn’t have the same number of frames.

I can think of a few different implementations that would allow me to do what I want:

- Generate one huge .aseprite file that would be the result of merging all .aseprite files together, then build the sprite sheet from it
- Reference all the .aseprite files that need to compose the sprite sheet, then generate it in one go
- Generate one sprite sheet per .aseprite file, then merge all the sprite sheets into a single one

This could be implemented in the CLI, but also in the UI if people think it'd be useful enough.

Thanks.

**Comment by thomasgoldstein** [comment]
Having played more with the CLI, it seems it's already supported. The CLI supports multiple paths, even though I'm not sure it's documented. I could not find this information on the CLI page (https://www.aseprite.org/docs/cli), but maybe I've overlooked it.

This command works just like the "Reference all the .aseprite files that need to compose the sprite sheet, then generate it in one go" option:

`aseprite --batch anim1.aseprite anim2.aseprite anim3.aseprite --trim --sheet output.png --data output.json`

Closing!
